project/src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -29,32 +29,43 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Set<Restaurant> restaurantSet = new HashSet<>();
     BufferedReader reader = request.getReader();
     String body = reader.readLine();
+    Map<Restaurant, Double> restaurantScores;
+    // TODO: handle case where list of restaurants is empty.
     try {
-      addRestaurantsToSet(body, restaurantSet);
+      JSONObject reqBody = new JSONObject(body);
+      restaurantScores = scoreRestaurants(
+          reqBody.getJSONArray("restaurants"), reqBody.getJSONObject("preferences"));
     } catch (JSONException e) {
       LOGGER.log(Level.WARNING, "Error parsing JSON: " + e.getMessage());
+      restaurantScores = new HashMap<>();
     }
```

Comment:
Is this needed? now that we're handling the exception inside scoreRestaurants and it doesnt throw it anymore

Reply:
I believe line 37 could still throw an exception

Reply:
ah, that would be a case where reqBody itself isnt usable so we cant do anything more after that -- I think lets make this: 
```
try {
    JSONObject reqBody = new JSONObject(body);
    Map<Restaurant, Double> restaurantScores = scoreRestaurants(
          reqBody.getJSONArray("restaurants"), reqBody.getJSONObject("preferences"));
     // TODO: handle empty restaurant set, maybe by response.println("no results found"):
     // Sort restaurant entries by highest score and write list of entries to the response.
    List<Map.Entry<Restaurant, Double>> sortedRestaurants =
        new ArrayList(restaurantScores.entrySet());
    sortedRestaurants.sort(Map.Entry.comparingByValue(Collections.reverseOrder()));
    response.getWriter().println(gson.toJson(sortedRestaurants));
} catch (JSONException e) {
  log("Error parsing request body: " + e.getMessage());
  // TODO: print user-friendly output to response
  // such as response.getWriter().println("error processing request");
}
```

Reply:
in other words -- do the remaining processing in the try block as if we catch an Exception while parsing the request body, nothing to do except showing the user some appropriate error message